x-pack/platform/plugins/shared/lens/public/react_embeddable/renderer/lens_embeddable_component.tsx

```diff
@@ -84,6 +130,17 @@ export function LensEmbeddableComponent({
         infoMessages={infoMessages}
         canEdit={canEdit}
       />
+      {isRuleFormVisible && canShowRuleForm && (
```

Comment:
@dej611 is the expert here but wouldn't this be cleaner if this logic was located outside the embeddable?

Reply:
I wanted to make sure this logic would work regardless of what context the vis was embedded in. If we're restricting rules to only dashboards then maybe we can move the code to dashboards instead.

Keeping it in the embeddable does future-proof it IMO, in case we want to enable rule creation from another context.

Reply:
I agree with @stratoula here.
I think this code does not belong here, rather should live within the `initializeAlertRules` and have a flow similar to the `onEdit`: somebody calls a `onAlertRuleCreation` or similar method and this should mount the flyout and handle the logic in there.
This has 2 benefits:
* handle the alerts logic within a single place
* avoid to bloat the embeddable component for everybody despite its usage of the alert feature.

Reply:
Where do we mount the flyout? Which React component do we render it in?

Reply:
Oh I see we want to use something like `x-pack/platform/plugins/shared/lens/public/react_embeddable/inline_editing/mount.tsx`

Reply:
Fixed in https://github.com/elastic/kibana/pull/217719/commits/d8670e8488cc9829d16b1956bfdccceca314dfe2